container: Used zonal cluster for network config test

wyardley opened the pull request:
- Use zonal cluster for `TestAccContainerNodePool_withNetworkConfig` test. Pick zone c (though a is used for most other zonal tests). This should result in 1/3 or 1/4 of the nodes being created for each of the multiple node pools in this test, which should improve both the speed and the reliability of this test
- Fix formatting

Noted while working on #14834

Note: I haven't been able to test this locally, both because of quota issues, and because of this error:
`Disabling pod cidr overprovision is not allowed for this project`
on my personal test project

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:none

```

slevenick commented:
I'm not sure why this error happened:
=== NAME  TestAccContainerNodePool_withConfidentialBootDisk
    bootstrap_iam_test_utils.go:119: [DEBUG] All bindings already present for project ci-project
=== NAME  TestAccContainerCluster_withConfidentialBootDisk
    bootstrap_iam_test_utils.go:119: [DEBUG] All bindings already present for project ci-project
panic: interface conversion: interface {} is *url.Error, not []interface {}

It shouldn't be related? I can retry

slevenick commented:
/gcbrun

wyardley commented on mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.tmpl:
```diff
@@ -4110,85 +4111,84 @@ resource "google_container_cluster" "cluster" {
-  location           = "us-central1"
-  cluster            = google_container_cluster.cluster.name
+  name       = "%s-manual"
+  location   = "us-central1-c"
```
@slevenick not sure if it would be best / cleaner to use `google_container_cluster.cluster.location` in all of these if it works? obviously there's already an implicit dependency already with the cluster name, but it would keep them consistent without having to declare a variable. 🤷 
Happy to update it to that if you think it would be better... I think it will work either way.

slevenick replied:
Yeah, that sounds reasonable. Then updating the location will be super simple in the future

wyardley replied:
Ok, updated. Trying a local test with the two node pools that were causing me issues commented out.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/5769f82b55f7799941cb5006e95535acd542bed3..746c13d1d483b84d2759506302ce353f1f0b37e7) ( 1 file changed, 42 insertions(+), 42 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/86eec1e7326926f8d2a52a91f6386672e640247a..5c5f4ffa38991712b695e76961c9d2751446b58c) ( 1 file changed, 42 insertions(+), 42 deletions(-))

modular-magician commented:
#### Tests analytics
Total tests: 121
Passed tests: 112
Skipped tests: 9
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>container</li>

</ul>

</blockquote>
</details>

🔴 Errors occurred during REPLAYING mode. Please fix them to complete your PR.

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14847/artifacts/76fd4af9-202b-42c9-96f1-44caa6d0a70f/build-log/replaying_test.log)

slevenick commented:
The error is still happening, but I'm sure it's entirely unrelated to your change. I'll try to investigate that

slevenick commented:
Ok my PR passed, so it actually does seem to be something in your change. I'm really not sure why. One possible fix is changing the test name so it thinks it's entirely new. Can you give that a try?

wyardley commented:
I rebased, updated the locations, and also put that extra newline back, though I don't think it's the issue. Also fixed an issue where the large nodepool already had `node_locations` set explicitly... left it there, but updated to zone c to match the cluster.
Trying to repro the replaying issue locally by commenting out the two nodepools that I was having issues with temporarily.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/513a9659d67f5b4b0127e3fcb4b1469c466ca83f..93a727aba66b8d9f5435d22c9dfc05c557d21b99) ( 1 file changed, 42 insertions(+), 41 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/87e56fcfd420902b3a33d55a049508c88066e1ec..01a8df96c522899013e4db01321e3f34a279737b) ( 1 file changed, 42 insertions(+), 41 deletions(-))

modular-magician commented:
#### Tests analytics
Total tests: 72
Passed tests: 60
Skipped tests: 11
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>container</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccContainerNodePool_withNetworkConfig</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccContainerNodePool_withNetworkConfig` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14847/artifacts/58536c22-e018-4293-ad86-52505cb8ed36/recording/TestAccContainerNodePool_withNetworkConfig.log)]


🔴 Tests failed when rerunning REPLAYING mode:
`TestAccContainerNodePool_withNetworkConfig` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14847/artifacts/58536c22-e018-4293-ad86-52505cb8ed36/build-log/replaying_build_after_recording/TestAccContainerNodePool_withNetworkConfig_replaying_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14847/artifacts/58536c22-e018-4293-ad86-52505cb8ed36/replaying_after_recording/TestAccContainerNodePool_withNetworkConfig.log)]


Tests failed due to non-determinism or randomness when the VCR replayed the response after the HTTP request was made.

Please fix these to complete your PR. If you believe these test failures to be incorrect or unrelated to your change, or if you have any questions, please raise the concern with your reviewer.


---






🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-14847/artifacts/58536c22-e018-4293-ad86-52505cb8ed36/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-14847/artifacts/58536c22-e018-4293-ad86-52505cb8ed36/recording) for each test
